packages/client/src/Views/PlayerUi/PlayerUi.tsx

```diff
@@ -65,48 +71,46 @@ const PlayerUi = ({ gameInfo, wargame, messageTypes, checkPasswordFail, loadData
 
   const handleCheckPassword = (pass: string): void => {
     const check = checkPassword(pass, messageTypes, currentWargame, allForces, dispatch)
-    if (check) setScreen('player')
+    if (check) setScreen(Room.player)
     else checkPasswordFail()
   }
 
-
-  if (screen === 'landing')
-  return <PlayerUiLandingScreen
-    gameInfo={gameInfo}
-    enterSerge={() => { setScreen('lobby') }}
-  />
-
-  if (screen === 'lobby') {
-    // TODO import type from PlayerUiLobby or move this function in to PlayerUiLobby
-    const roleOptions = (): ({ name: string, roles: Role[] })[] => allForces.map(
-      force => ({name: force.name, roles: force.roles})
-    )
-
-    return <PlayerUiLobby
-      wargameList={wargame.wargameList}
-      roleOptions={roleOptions()}
-      checkPassword={handleCheckPassword}
-    />
-  }
-
-  if (selectedForce) {
-
-    if (wargameInitiated) {
-      const setStorageKey = (): string => `${wargameTitle}-${selectedForce.uniqid}-${selectedRole}-tourDone`
-      return <GameChannelsWithTour
-        storageKey={setStorageKey()}
-        tourIsOpen={tourIsOpen}
+  // show the relevant screen
```

Comment:
@lilitkarapetyan - I've refactored the previous code to this switch structure. The code works, but I'd appreciate you checking the implementation is ok.